Add missing documentation for 1.59 (#735)
eclipse-theia/theia-website: src/docs/user_ai.md

This section provides an overview of the currently available agents in the Theia IDE. Agents marked as “Chat Agents” are available in the global chat, while others are directly integrated into UI elements, such as code completion. You can configure and deactivate agents in the AI Configuration view.

### Theia Coder (Chat Agent)

> An AI assistant is designed to assist software developers. This agent can access the users workspace, it can get a list of all available files and folders and retrieve their content. Futhermore, it can suggest modifications of files to the user. It can therefore assist the user with coding tasks or other tasks involving file changes. See the dedicated [Theia Coder Documentation](/docs/theia_coder) for more details.

Reviewer: ```suggestion
An AI assistant designed to assist software developers. This agent can access the users workspace, it can get a list of all available files and folders and retrieve their content. Furthermore, it can suggest modifications of files to the user. It can therefore assist the user with coding tasks or other tasks involving file changes. See the dedicated [Theia Coder Documentation](/docs/theia_coder) for more details.
```

This agent is aware of all commands available in the Theia IDE and the current tool the user is working with. Based on the user request, it can find the appropriate command and let the user execute it.

### Architect (Chat Agent)

> An AI assistant is designed to assist software developers. This agent can access the users workspace, it can get a list of all available files and folders and retrieve their content. It cannot modify files. It can therefore answer questions about the current project, project files and source code in the workspace, such as how to build the project, where to put source code, where to find specific code or configurations, etc.

Reviewer: ```suggestion
An AI assistant designed to assist software developers. This agent can access the users workspace, it can get a list of all available files and folders and retrieve their content. It cannot modify files. It can therefore answer questions about the current project, project files and source code in the workspace, such as how to build the project, where to put source code, where to find specific code or configurations, etc.
```

<img src="../../general-chat.png" alt="General AI Chat in the Theia IDE" style="max-width: 525px">

Some agents produce special results, such as buttons (shown in the screenshot above) or code that can be directly inserted. 

### Agent Pinning
The *Agent Pinning* feature, reduces the need for repeated agent references.  
  
- When you mention an agent in a prompt and no agent is pinned, the mentioned agent is automatically pinned.  
- If an agent is already pinned, mentioning a different agent will **not** change the pinned agent. Instead, the newly mentioned agent will be used **only** for that specific prompt.  
- You can manually unpin an agent through the chat interface if needed.  

> - A new chat session can now be started directly with a pinned agent.

Reviewer: ```suggestion
```

There is no command anymore to start a new chat with a pinned agent. Instead the @ symbol triggers the agent selection and then the rules above just apply. So creating a new chat with a pinned agent is essentially now starting a new chat and selecting an agent in the first message.
